### DISCUSSION TITLE
PUGIXML AND MULTITHREADING

### DISCUSSION THREAD

**Discussion by Alexey-cpu** [zeux/pugixml#646] [Q&A] [answered]
I use pugixml in my project and I have a question concerning multithreading: is it safe to create TWO DIFFERENT xml documents and read/write/modify these documents from different threads ?

**Comment by zeux** [comment]
Yes, see https://pugixml.org/docs/manual.html#dom.thread.

**Reply by Alexey-cpu** [reply]
Ok.
But, I have read the notes about thread-safety guarantees and I've been confused by this note:

![image](https://github.com/user-attachments/assets/f7a06f94-48c2-48ee-ab76-0f89a7c4f7f5)

From this note It seems like I cannot modify TWO DIFFERENT trees at a time from different threads.
May be, I have misunderstood it. Could you please clarify this moment ?

**Reply by zeux** [reply, marked as answer]
Yeah maybe this should be reworded. Each thread exclusively accessing its own document is safe; you can have 5 documents and 5 threads, as long as each thread only reads/writes a separate document. "tree" in the documentation refers to a document as well.

**Reply by Alexey-cpu** [reply]
Ok, now it's clear.
Thank you very much.
